File: LibreriaV3.1/LibreriaV3.1/Persistencia/IAcceso.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LibreriaV3._1.Persistencia
{
    interface IAcceso<obj>
    {
        bool Insertar(obj obj);
        bool Borrar(object objeto);
        Object Buscar(Type clase, String nombre);
        List<object> Obtener(Type clase);
        bool Modificar(string nombre, obj obj);
        bool BorradoVirtual(object objeto);
        List<object> Buscar(Type clase, string campo, string busqueda);
    }
}

File: LibreriaV6.1/src/Persistencia/IAcceso.java
package Persistencia;

import javax.naming.NamingException;
import java.io.IOException;
import java.lang.reflect.InvocationTargetException;
import java.sql.SQLException;
import java.util.ArrayList;

/**
 * Created by Juangra on 07/11/2016.
 */
public interface IAcceso <obj> {

    boolean insertar(ArrayList<obj> list) throws Exception;
    boolean borrar(Object object) throws Exception;
    Object buscar(Class clase, String nombre) throws Exception;
    ArrayList<obj> obtener(Class clase) throws Exception;
    boolean modificar(String nombre,obj obj) throws Exception;


}
